pkg/scheduler/actions/backfill/backfill.go

```diff
@@ -68,6 +74,101 @@ func (alloc *backfillAction) Execute(ssn *framework.Session) {
 			}
 		}
 	}
+
+	if len(alloc.arguments) == 0 {
+		return
+	}
+
+	backfillNonBestEffortJobsEnabled, err := strconv.ParseBool(alloc.arguments[conf.BackfillFlagName])
+	if err != nil {
+		glog.V(3).Infof("backfilling non best effort jobs is disabled")
+		return
+	}
+
+	if backfillNonBestEffortJobsEnabled {
+		// Collect back fill candidates
+		backFillCandidates := make([]*api.JobInfo, 0, len(ssn.Jobs))
+		for _, job := range ssn.Jobs {
+			if !ssn.BackFillEligible(job) {
+				continue
+			}
+			backFillCandidates = append(backFillCandidates, job)
+		}
+
+		// Release resources allocated to unready top dog jobs so that
+		// we can back fill more jobs in the next step.
+		resourceReleased := false
+		for _, job := range ssn.Jobs {
```

Comment:
can we merge this with above loop?

Reply:
merged.

if a job is not ready even if by borrowing resources from backfilled jobs, it will release its allocated resources  because it won't be ready anyway.

Reply:
may consider removing the candidates selection all together. WIP.